src/WebView.php

```diff
@@ -138,17 +140,21 @@ public function head(): void
      */
     public function beginBody(): void
     {
-        echo sprintf(self::PLACEHOLDER_BODY_BEGIN, $this->getPlaceholderSignature());
-        $this->eventDispatcher->dispatch(new BodyBegin($this->getViewFile()));
+        if (is_string($viewFile = $this->getViewFile())) {
+            echo sprintf(self::PLACEHOLDER_BODY_BEGIN, $this->getPlaceholderSignature());
+            $this->eventDispatcher->dispatch(new BodyBegin($viewFile));
+        }
     }
 
     /**
      * Marks the ending of an HTML body section.
      */
     public function endBody(): void
     {
-        $this->eventDispatcher->dispatch(new BodyEnd($this->getViewFile()));
-        echo sprintf(self::PLACEHOLDER_BODY_END, $this->getPlaceholderSignature());
+        if (is_string($viewFile = $this->getViewFile())) {
```

Comment:
How could it be non-string?

Reply:
> It could be false when rendering from View directly
> You can reproduce this behavior with the given method of the controller
> ` public function index(WebView $view, DataResponseFactoryInterface $responseFactory): ResponseInterface { return $responseFactory->createResponse($view->render('index')); }`

same, but probably only when called this method alone